src/Computation/PooledSolver.cs

```diff
@@ -58,23 +60,18 @@ public bool IsSatisfiable(IValue assertion)
         };
     }
 
-    public BigInteger GetSingleValue(IValue variable)
+    public BigInteger GetSingleValue(IValue value)
     {
-        using var bitVector = variable.AsBitVector(this);
-        using var simplified = bitVector.Simplify();
+        var constant = GetExampleValue(value);
 
-        var value = simplified.IsNumeral
-            ? ((BitVecNum) simplified).BigInteger
-            : throw new IrreducibleSymbolicExpressionException();
-
-        return value == GetExampleValue(variable)
-            ? value
+        return IsSatisfiable(NotEqual.Create(value, ConstantUnsigned.Create(value.Size, constant)))
```

Comment:
Shouldn't these branches be the other way around? I.e if this is false because there aren't any other constants it can be then return the constant.

Reply:
This is why one should always run tests.